app/models/user.rb

```diff
@@ -3,9 +3,12 @@
 class User < ApplicationRecord
   has_secure_password
   has_secure_token :auth_token
+  has_secure_token :email_confirmation_token
 
-  validates :email, presence: true, uniqueness: true, length: 1..100
+  validates :email, presence: true, uniqueness: true, length: 1..100, format: { with: URI::MailTo::EMAIL_REGEXP }
+  validates :unconfirmed_email, presence: true, uniqueness: true, length: 1..100, format: { with: URI::MailTo::EMAIL_REGEXP }, allow_blank: true
```

Comment:
We might not want `presence: true` here? Needs more thought, I'll follow up on this later.

Reply:
We concluded we definitely do not want the presence validation here.